### ISSUE TITLE
Problems on 4.7.x versions

### ISSUE THREAD

**Issue by francescbassas** [ixiam/com.ixiam.payment.redsys#17]
I'm having issues with IPN processing with the Redsys Payment Processor on a 4.7.12 installation.

After completing a contribution no mail notifications are sending to the contributor and the status of the contribution stays "Pending". If I check logs I can see errors like:

```
Nov 09 12:20:41  [info] Could not find payment processor for contribution record: 86

Nov 09 12:20:41  [info] Redsys Validation failed
```

As you can see, the problems I have are very similar with the https://issues.civicrm.org/jira/browse/CRM-17980. Alex C suggest to convert the 'payment' type extension to 'module' type (on info.xml).  ~~I'm not sure if this is possible without breaking anything or if it is necessary to rewrite the extension from 'zero' or at least doing some extension upgrade.~~ But seems that this isn't our case because info.xml already has defined type as module.

**Comment by ixiam** [comment]
Hi @francescbassas , we haven't tested the extension in CiviCRM 4.7. 
If you like to test it and fix it, we can open a new branch of the repo (4.7) for this case

**Comment by francescbassas** [comment]
@ixiam I got it (or so I think)! You can view my PR https://github.com/ixiam/com.ixiam.payment.redsys/pull/18. Probably this bug affects other issues like https://github.com/ixiam/com.ixiam.payment.redsys/issues/15.

**Comment by ixiam** [comment]
@francescbassas I think this fix is not backwards compatible with CiviCRM <= 4.6, 
so we have 2 options at this point:

1. Create 2 branches of the extension (4.4 / 4.5 / 4.6 & 4.7), with different tags, and move away from master branch.
2. Add an *if* statement above your change, checking version and if it's <= 4.6 use old source code and if it's = 4.7 use your new code. 

This code worked for <=4.6, not sure for 4.7:
`
if (CRM_Utils_System::version() <= 4.6) {
}
else {
}`


I think 2. it's the best approach, so we keep using one version of the extension, but we will need to test it in the different CiviCRM version to be sure it works for all of them

**Comment by francescbassas** [comment]
B sounds good for me. But maybe we can wait to @helenagarcia90 feedback on issue https://github.com/ixiam/com.ixiam.payment.redsys/issues/15 to ensure that this bugs only affects to 4.7 CiviCRM versions.

**Comment by francescbassas** [comment]
I'm back. We do not have news from @helenagarcia90? There is no bug related with this suspicious line of code on 4.6 versions?

https://github.com/ixiam/com.ixiam.payment.redsys/blob/master/CRM/Core/Payment/Redsys.php#L234

Apparently assigns a payment processor type ID to a payment processor ID

`$paymentProcessorID = CRM_Core_DAO::getFieldValue('CRM_Financial_DAO_PaymentProcessorType', $this->_processorName, 'id', 'name');`

Otherwise we can proceed to apply the @sluc23 B solution.

**Comment by sluc23** [comment]
Hello @francescbassas , we pushed in develop branch a fix for 4.7 in https://github.com/ixiam/com.ixiam.payment.redsys/commit/5045da90a7e29a7eccfe27ec4e4bad140ea94b36 

We partially tested it in 4.7 and looks good. Please keep on testing and once it's stable we can push a new release

**Comment by francescbassas** [comment]
Thanks @sluc23. Works as expected and @elcapo fix looks more clean. For me it's ok.

**Comment by sluc23** [comment]
fixed in v1.8
